docs/static/schemas/draft/2-0-0/application.schema.json

```diff
@@ -0,0 +1,1564 @@
+                  ]
+                },
+                "controller_predicate": {
+                  "title": "Controller Predicate Condition",
+                  "description": "A condition depending on the runtime state of a controller predicate",
+                  "type": "object",
+                  "additionalProperties": false,
+                  "properties": {
+                    "controller": {
+                      "$ref": "#/$defs/events/properties/call_service/oneOf/2"
+                    },
+                    "hardware": {
+                      "$ref": "#/$defs/events/properties/call_service/oneOf/2"
+                    },
+                    "predicate": {
+                      "$ref": "#/$defs/events/properties/call_service/oneOf/2"
+                    }
+                  },
+                  "required": [
+                    "controller",
+                    "hardware",
+                    "predicate"
+                  ]
+                },
+                "component_state": {
+                  "title": "Component State Condition",
+                  "description": "A condition depending on the runtime state of a component",
+                  "type": "object",
+                  "additionalProperties": false,
+                  "properties": {
+                    "component": {
+                      "$ref": "#/$defs/events/properties/call_service/oneOf/2"
+                    },
+                    "state": {
+                      "title": "Component State",
+                      "description": "The runtime state of the component",
+                      "type": "string",
+                      "enum": [
+                        "unloaded",
+                        "loaded",
+                        "unconfigured",
+                        "inactive",
+                        "active",
+                        "finalized"
+                      ]
+                    }
+                  },
+                  "required": [
+                    "component",
+                    "state"
+                  ]
+                },
+                "controller_state": {
+                  "title": "Controller State Condition",
+                  "description": "A condition depending on the runtime state of a controller",
+                  "type": "object",
+                  "additionalProperties": false,
+                  "properties": {
+                    "controller": {
+                      "$ref": "#/$defs/events/properties/call_service/oneOf/2"
+                    },
+                    "hardware": {
+                      "$ref": "#/$defs/events/properties/call_service/oneOf/2"
+                    },
+                    "state": {
+                      "title": "Controller State",
+                      "description": "The runtime state of the controller",
+                      "type": "string",
+                      "enum": [
+                        "unloaded",
```

Comment:
Just to be sure, are those really all the states that the DSE can know about controllers? no inactive or configured?